File: 2022/rust/aoc2022/src/bin/01.rs
use itertools::Itertools;

fn main() {
    let elves = aoc2022::input_as_string(1)
        .split("\n\n")
        .map(|l| l.lines().map(|i| i.parse::<u32>().unwrap()).sum::<u32>())
        .sorted()
        .rev()
        .collect_vec();

    let max = elves.first().unwrap();
    let top3: u32 = elves.iter().take(3).sum();

    println!("part1: {}", max);
    println!("part2: {}", top3);
}

File: 2022/python/01.py
elves = [sum(x) for x in [[int(l) for l in g.split('\n')] for g in open("input/01").read().split("\n\n")]]
print('p1: {}'.format(max(elves)))
print('p2: {}'.format(sum(sorted(elves, reverse=True)[:3])))